src/matgl/models/_m3gnet.py

```diff
@@ -251,35 +259,60 @@ def forward(
                 edge_feat,
             )
             edge_feat, node_feat, state_feat = self.graph_layers[i](g, edge_feat, node_feat, state_feat)
+            fea_dict[f"gc_{i+1}"] = {"node_feat": node_feat, "edge_feat": edge_feat, "state_feat": state_feat}
         g.ndata["node_feat"] = node_feat
         g.edata["edge_feat"] = edge_feat
         if self.is_intensive:
-            node_vec = self.readout(g)
-            vec = torch.hstack([node_vec, state_feat]) if self.include_states else node_vec  # type: ignore
-            output = self.final_layer(vec)
+            field_vec = self.readout(g)
+            readout_vec = torch.hstack([field_vec, state_feat]) if self.include_states else field_vec  # type: ignore
+            fea_dict["readout"] = readout_vec
+            output = self.final_layer(readout_vec)
             if self.task_type == "classification":
                 output = self.sigmoid(output)
         else:
             g.ndata["atomic_properties"] = self.final_layer(g)
+            fea_dict["readout"] = g.ndata["atomic_properties"]
             output = dgl.readout_nodes(g, "atomic_properties", op="sum")
+        fea_dict["final"] = output
+        if return_all_layer_output:
+            return fea_dict
         return torch.squeeze(output)
 
-    def predict_structure(
+    def featurize_structure(
         self,
         structure,
         state_feats: torch.Tensor | None = None,
         graph_converter: GraphConverter | None = None,
+        output_layers: list | None = None,
     ):
-        """Convenience method to directly predict property from structure.
+        """Convenience method to featurize a structure with M3GNet model.
 
         Args:
             structure: An input crystal/molecule.
-            state_feats (torch.tensor): Graph attributes
+            state_feats (torch.tensor): Graph attributes.
             graph_converter: Object that implements a get_graph_from_structure.
+            output_layers: List of names for the layer of GNN as output. Choose from "bond_expansion", "embedding",
+                "three_body_basis", "gc_1", "gc_2", "gc_3", "readout", and "final". By default, all M3GNet layer
+                outputs are returned.
 
         Returns:
-            output (torch.tensor): output property
+            output (dict): M3GNet intermediate and final layer outputs for a structure.
         """
+        allowed_output_layers = [
```

Comment:
This cannot be hard coded. So if you have more than three GCLs or less than 3 GCLs, this code fails. Think about what kind of code you are creating. Are we saying M3GNet ALWAYS have 3 GCLs?

Reply:
I will address this in next commit.